app/services/GetCalculationDetailsService.scala

```diff
@@ -17,28 +17,37 @@
 package services
 
 import connectors.httpParsers.CalculationDetailsHttpParser.CalculationDetailResponse
-import connectors.{CalculationDetailsConnectorLegacy, GetCalculationListConnector}
+import connectors.{CalculationDetailsConnector, CalculationDetailsConnectorLegacy, GetCalculationListConnector}
 import models.{DesErrorBodyModel, DesErrorModel}
 import play.api.http.Status.NO_CONTENT
 import uk.gov.hmrc.http.HeaderCarrier
+import utils.TaxYear
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class GetCalculationDetailsService @Inject()(calculationDetailsConnector: CalculationDetailsConnectorLegacy,
+class GetCalculationDetailsService @Inject()(calculationDetailsConnectorLegacy: CalculationDetailsConnectorLegacy,
+                                             calculationDetailsConnector: CalculationDetailsConnector,
                                              listCalculationDetailsConnector: GetCalculationListConnector) (implicit ec: ExecutionContext) {
 
   def getCalculationDetails(nino: String, taxYear: Option[String])(implicit hc: HeaderCarrier):Future[CalculationDetailResponse] = {
     listCalculationDetailsConnector.calcList(nino, taxYear).flatMap {
       case Right(listOfCalculationDetails) if(listOfCalculationDetails.isEmpty) =>
         Future.successful(Left(DesErrorModel(NO_CONTENT, DesErrorBodyModel.parsingError)))
       case Right(listOfCalculationDetails) =>
-        getCalculationDetailsByCalcId(nino, listOfCalculationDetails.head.calculationId)
+        getCalculationDetailsByCalcId(nino, listOfCalculationDetails.head.calculationId, taxYear)
       case Left(desError) => Future.successful(Left(desError))
     }
   }
 
-  def getCalculationDetailsByCalcId(nino: String, calcId: String)(implicit hc: HeaderCarrier): Future[CalculationDetailResponse] = {
-    calculationDetailsConnector.getCalculationDetails(nino, calcId)
+  def getCalculationDetailsByCalcId(nino: String, calcId: String, taxYear: Option[String])(implicit hc: HeaderCarrier): Future[CalculationDetailResponse] = {
+
+    TaxYear.convert(taxYear) match {
+      case _ if taxYear.isEmpty => calculationDetailsConnectorLegacy.getCalculationDetails(nino, calcId)
```

Comment:
we might need to assume that this is and error in the future, as taxYear should be mandatory regardless what API we are calling; but we can leave this change for later